packages/visual-editor/src/internal/components/InternalLayoutEditor.tsx

```
@@ -253,6 +254,29 @@ export const InternalLayoutEditor = ({
     } as Config;
   }, [puckConfig, i18n.language]);
 
+  // Resolve all data and slots when the document changes
+  // Implemented as an override so that the getPuck hook is available
+  const reloadDataOnDocumentChange = React.useCallback(
+    (props: { children: React.ReactNode }): React.ReactElement => {
+      const getPuck = useGetPuck();
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)